lib/user_lookup.rb

```diff
@@ -12,66 +19,46 @@ def [](user_id)
   end
 
   def primary_groups
-    @primary_groups ||= begin
-      hash = {}
-      users.values.each do |u|
-        if u.primary_group_id
-          hash[u.id] = groups[u.primary_group_id]
-        end
+    @primary_groups ||= users
+      .values
+      .filter(&:primary_group_id)
```

Comment:
I don't feel like this improves readability that much that we should make the implementation less efficient by doing 2 loops.

Reply:
Imo using `being/end` block and a helper variable to avoid using a functional chain is an unnecessary micro-optimization. The first iterator reduced the number of elements in collection so it's not like it loops twice over all elements.

Reply:
I understand where you’re coming from but I don’t feel very strongly that this refactor improves readability that much.